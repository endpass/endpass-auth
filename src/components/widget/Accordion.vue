<template>
  <section class="accordion">
    <slot name="control" />
    <section
      v-show-slide="!collapsed"
      :style="{ maxHeight: maxHeight }"
      class="accordion-content"
    >
      <slot />
    </section>
  </section>
</template>

<script>
export default {
  name: 'Accordion',

  props: {
    maxHeight: {
      type: String,
      default: 'none',
    },

    collapsed: {
      type: Boolean,
      default: true,
    },
  },
};
</script>

<style lane="postcss">
.accordion-content {
  overflow-y: auto !important;
}
</style>
