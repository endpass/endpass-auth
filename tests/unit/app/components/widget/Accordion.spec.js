import { shallowMount, createLocalVue } from '@vue/test-utils';
import VShowSlide from 'v-show-slide';
import Accordion from '@/components/widget/Accordion.vue';

const localVue = createLocalVue();

localVue.use(VShowSlide);

describe('Accordion', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(Accordion, {
      localVue,
    });
  });

  it('should correctly render', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
