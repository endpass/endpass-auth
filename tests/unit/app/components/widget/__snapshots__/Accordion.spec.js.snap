// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion should correctly render 1`] = `
<section class="accordion">
  <section class="accordion-content" style="max-height: none; height: 0px; overflow: hidden;"></section>
</section>
`;
